### PR TITLE
liquidator: add tryResolveBankruptUser

### PR DESCRIPTION
**Pull request by wphan**

### CODE REVIEW

File: src/bots/pnlSettler.ts [hunk 1]
```diff
@@ -158,26 +158,41 @@ export class PnlSettlerBot implements Bot {
...
-								users: [userData],
-								marketIndex: settleePosition.marketIndex,
-							});
-						}
+					if (unsettledPnl.gt(MIN_PNL_TO_SETTLE)) {
+						continue;
+					}
+
+					// only settle user pnl if they have enough collateral
+					if (
```
Comment (crispheaney) [inline src/bots/pnlSettler.ts:168]: naisu

File: src/bots/liquidator.ts [hunk 2]
```diff
@@ -217,17 +219,147 @@ export class PerpLiquidatorBot implements Bot {
...
+		// temp on old sdk impl.
+		for (const position of chUserToCheck.getUserAccount().spotPositions) {
+			if (!isVariant(position.balanceType, 'borrow')) {
+				// not possible to resolve non-borrow markets
+				continue;
+			}
+			const spotMarket = this.clearingHouse.getSpotMarketAccount(
+				position.marketIndex
+			);
+			const positionBorrowAmount = getTokenAmount(
```
Comment (crispheaney) [inline src/bots/liquidator.ts:273]: think you want to do getTokenValue here so the borrow is quoted in usd

File: src/bots/liquidator.ts [hunk 3]
```diff
@@ -217,17 +219,147 @@ export class PerpLiquidatorBot implements Bot {
...
+			logger.warn(
+				`UserAcc: ${userKey.toBase58()}, Auth: ${auth} is bankrupt but no resolvable market`
+			);
+		} else if (perpIdx > -1 && spotIdx === -1) {
+			marketTypeToResolve = 'perp';
+		} else if (perpIdx === -1 && spotIdx > -1) {
+			marketTypeToResolve = 'spot';
+		} else {
+			// TODO: are perpLoss and spotLoss precisions correct to compare here?
+			if (perpLoss.abs().gt(spotLoss.abs())) {
```
Comment (crispheaney) [inline src/bots/liquidator.ts:311]: ok to do if you cast it into usd, but you may just want to try to resolve them both at the same time? why'd you want to do it one at a time based on size?

Comment (wphan) [reply src/bots/liquidator.ts:311]: probably got a bit mixed up with how liquidation worked, so thought i should resolve 1 position, then check if the account is still bankrupting and resolve the next. going to just resolve all markets with loss as Z pointed out too

**Pull request merged**